### ISSUE TITLE
Gradle 4.3-RC4 fails to resolve dependency where 4.2.1 succeeds

### ISSUE THREAD

**Issue by tetsuo55** [gradle/gradle#3299]
Our project uses a dependency that calls another dependency
https://github.com/substratum/substratum/blob/release/app/build.gradle#L145

this works fine in 4.2.1 however in 4.3 i get the following error

```
Unable to resolve dependency for ':app@debug/compileClasspath': Could not resolve com.github.asyl.animation:arcanimator:1.0.0.

Could not resolve com.github.asyl.animation:arcanimator:1.0.0.
Required by:
    project :app > com.gordonwong:material-sheet-fab:1.2.1
 > Could not resolve com.github.asyl.animation:arcanimator:1.0.0.
    > Could not get resource 'https://jitpack.io/com/github/asyl/animation/arcanimator/1.0.0/arcanimator-1.0.0.pom'.
          > Could not HEAD 'https://jitpack.io/com/github/asyl/animation/arcanimator/1.0.0/arcanimator-1.0.0.pom'. Received status code 401 from server: Unauthorized
```

### Expected Behavior
gradle should compile without errors

### Current Behavior
gradle fails to compile because of the resolve issues

### Steps to Reproduce (for bugs)
checkout and build https://github.com/substratum/substratum with gradle 4.3

### Your Environment
android studio 3.1 on windows

**Comment by eskatos** [comment]
[4.2.1 build scan](https://scans.gradle.com/s/4t5mwwnqayhqo) success 
[4.3-rc-4 build scan](https://scans.gradle.com/s/77d4x3xgbnjzc) failure

**Comment by bmuschko** [comment]
> Received status code 401 from server: Unauthorized

That sounds more like you have an authentication issue of some sort though I am surprised that this would happen with Jitpack.

I tried to reproduce this with a simple build script but couldn't even resolve the dependency `material-sheet-fab` from Jitpack with Gradle 4.2.1 or 4.3-RC4.

```
apply plugin: 'base'

repositories {
    maven { url "https://jitpack.io" }
}

configurations {
    myConf
}

dependencies {
    myConf 'com.gordonwong:material-sheet-fab:1.2.1'
}

task copyLibs(type: Copy) {
    from configurations.myConf
    into "$buildDir/libs"
}
```

Output is:

```
> Could not find com.gordonwong:material-sheet-fab:1.2.1.
  Searched in the following locations:
      https://jitpack.io/com/gordonwong/material-sheet-fab/1.2.1/material-sheet-fab-1.2.1.pom
      https://jitpack.io/com/gordonwong/material-sheet-fab/1.2.1/material-sheet-fab-1.2.1.jar
  Required by:
      project :
```

**Comment by eskatos** [comment]
I would say that @tetsuo55 got bit by [this 4.3 breaking change](https://docs.gradle.org/release-candidate/release-notes.html#avoid-checking-other-repositories-when-dependency-resolution-in-one-repository-fails)
jitpack answers 401 to 4.2.1 too, but 4.2.1 falls back to other declared repositories.

**Comment by bmuschko** [comment]
@eskatos You might be right. @tetsuo55 Do you even need to resolve dependencies from Jitpack in your project?

**Comment by tetsuo55** [comment]
We need to have those add ons working. I will have to ask other devs about the need for jitpack specifically

**Comment by bmuschko** [comment]
I guess you could try to remove the Jitpack repository and run `./gradlew build --refresh-dependencies` to see if it is actually needed.

**Comment by tetsuo55** [comment]
i tried that and we do have an actual jitpack dependency,
How do you suggest i format the gradle file to prevent the accidental lookup in jitpack?

the dependency seems to be com.github.recruit-lifestyle:FloatingView:2.2.3-beta1'

**Comment by bmuschko** [comment]
I'd suggest you order the repositories in a way so that Jitpack is checked last [here](https://github.com/substratum/substratum/blob/release/build.gradle#L17-L23). As a consequence the dependency is downloaded from JCenter which works fine.

```
allprojects {
    repositories {
        google()
        jcenter()
        maven { url "https://jitpack.io" }
    }
}
```

There are [similar issues reported for Jitpack](https://github.com/jitpack/jitpack.io/issues/1916). Please open an issue with Jitpack for the module you are referencing.

In Gradle 4.3 we changed the behavior as [described in the release notes](https://docs.gradle.org/release-candidate/release-notes.html#avoid-checking-other-repositories-when-dependency-resolution-in-one-repository-fails). The failure you are seeing is expected if Gradle fails to resolve from the first repository.

**Comment by edeandrea** [comment]
I seem to have somewhat similar behavior. In Gradle 4.2.1 my build is fine, but with Gradle 4.3 my build fails with

```
FAILURE: Build failed with an exception.

* Where:
Build file '/Users/edeandre/workspaces/Stuff/aig-gradle/build.gradle' line: 13

* What went wrong:
Error resolving plugin [id: 'com.jfrog.artifactory', version: '4.4.9']
> Could not GET 'https://plugins.gradle.org/api/gradle/4.3/plugin/use/com.jfrog.artifactory/4.4.9'.
   > Connect to plugins.gradle.org:443 [plugins.gradle.org/104.16.174.166, plugins.gradle.org/104.16.171.166, plugins.gradle.org/104.16.175.166, plugins.gradle.org/104.16.173.166, plugins.gradle.org/104.16.172.166, plugins.gradle.org/2400:cb00:2048:1:0:0:6810:afa6, plugins.gradle.org/2400:cb00:2048:1:0:0:6810:aba6, plugins.gradle.org/2400:cb00:2048:1:0:0:6810:aea6, plugins.gradle.org/2400:cb00:2048:1:0:0:6810:aca6, plugins.gradle.org/2400:cb00:2048:1:0:0:6810:ada6] failed: No route to host (connect failed)

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
```

This is my `plugins` block at the very top of my build.gradle:

```groovy
plugins {
	id 'java-gradle-plugin'
	id 'groovy'
	id 'maven-publish'
	id 'jacoco'
	id 'build-dashboard'
        id 'eclipse'
	id 'com.jfrog.artifactory' version '4.4.9'
	id "org.sonarqube" version "2.5"
}
```

I am not overriding anything to do with the plugin repositories. In fact, my project does not have a `settings.gradle` file.

**Comment by FrancescoCapari** [comment]
This needs to be reopened and resolved ASAP, otherwise Gradle 4.3 is pretty much unusable. After upgrading to 4.3, none of our several Android projects are not able to resolve dependencies to libraries, mainly those from small developers that are in their custom repositories and not pushed to the big ones. Which should not be required as a change "Avoid checking other repositories when dependency resolution in one repository fails" forces to. This change is IMHO the worst and most breaking change ever introduced to Gradle.

**Comment by pioterj** [comment]
@FrancescoCapari Thanks for bringing this up. The problem should be resolved in Gradle 4.3.1.

**Comment by bmuschko** [comment]
@tetsuo55 With Gradle 4.4 we will treat dependency resolution handling for HTTP 5xx error codes the same way as we do for connection and socket timeouts (do not ask subsequent repositories). We'd love for you to try it out and provide us with feedback to avoid any potential issues in the final release. You can try out a snapshot distribution here: https://services.gradle.org/distributions-snapshots/gradle-4.4-20171112235945+0000-bin.zip. The actual change can be found here if you are interested: https://github.com/gradle/gradle/pull/3412. Thanks for your support! Let us know if you have any questions.

**Comment by lorenzopolidori** [comment]
I have an Android project that fetches its dependencies from multiple repositories, including small independent developers' maven repos. Everything worked fine with gradle 4.1 (android plugin 3.0) until I upgraded to gradle to 4.4 (android gradle plugin 3.1). 

Gradle now cannot resolve dependencies properly since, if a dependency in a small independent maven repo is not found in jcenter or google maven, or viceversa, if an android dependency is not found in the small repo, the build fails. I have tried many different combinations and ordering but the result is always the same. 

I agree with @FrancescoCapari, gradle 4.3+ seems to be unusable with projects that have dependencies from independent small repos.
